packages/ses/src/commons.js

```diff
@@ -177,18 +177,32 @@ export const arraySort = uncurryThis(arrayPrototype.sort);
 export const iterateArray = uncurryThis(arrayPrototype[iteratorSymbol]);
 //
 export const arrayBufferSlice = uncurryThis(arrayBufferPrototype.slice);
+const { asUintN: bigIntAsUintN } = BigInt;
+const { structuredClone } = globalThis;
 export const arrayBufferTransferToFixedLength =
   // @ts-expect-error absent from Node 20, which we still support
   arrayBufferPrototype.transferToFixedLength
     ? // @ts-expect-error absent from Node 20, which we still support
       uncurryThis(arrayBufferPrototype.transferToFixedLength)
-    : (arrayBuffer, newLength = arrayBuffer.byteLength) =>
+    : (arrayBuffer, newLength = arrayBuffer.byteLength) => {
         // There is no `transferToFixedLength` on Node 20, which we still support.
-        // In that case, there is no way just within the language to detach an
-        // ArrayBuffer, so the best we can do is emulate it using `slice`.
-        // Unfortunately, this leaves the original non-detached.
-        arrayBufferSlice(arrayBuffer, 0, newLength);
-//
+        // In that case, we use `slice` to get a fixed-length copy and,
+        // if present, the WHATWG HTML `structuredClone` to detach the original.
+
+        // `slice` accepts negative arguments but `transferToFixedLength` does not...
+        // get at the underlying ToIndex operation through `BigInt.asUintN`
+        // (avoiding the redundant allocation of e.g. `ArrayBuffer(newLength)`)
+        // and ToNumber through unary `+` (rather than `Number(newLength)`,
+        // which fails to reject BigInts).
+        newLength = +newLength;
+        bigIntAsUintN(newLength, 0n);
+
+        const copied = arrayBufferSlice(arrayBuffer, 0, newLength);
```

Comment:
This slice should not be present at all if `structuredClone` doesn't exist:
- a `transfer` method is meant to detach the original
- a `slice` is not needed even after transfer since the original cannot be resizable

We should only use a `slice` if there is no way to detach the original, which becomes a deficiency of the shim.

Reply:
> This slice should not be present at all if `structuredClone` doesn't exist:

I don't understand this. Seems inconsistent with 

> We should only use a `slice` if there is no way to detach the original, which becomes a deficiency of the shim.

which I do understand and agree with. In particular, if there is no `transfer`, `transferToFixed`, or `structuredClone`, then our shim should use `slice` and have this deficiency on those platforms. Yes?

Reply:
In a platform with `structuredClone` but without `transfer` or `transferToFixedLength`, I can transfer and detach without `slice` if the length doesn't need to change. But if the length does need to change, then I'd still need to use `slice`, yes?

Reply:
In what case does the length need to change?

Reply:
PTAL. I'm resolving because I think this is done, but please unresolve if you disagree.

Reply:
> In what case does the length need to change?

Argh. I see your point! Unresolving. I need to see if I can resize, and do that instead of `slice` when `resize` is possible.

Reply:
Note that there is no `resize` on Node <= 18.

Reply:
The ArrayBuffer made by `structuredClose` is not `resizable`, so that plan wouldn't have worked. I added a comment to that effect and continue to use `slice`. I don't see any other choice. Resolving again.